src/index.ts

```diff
@@ -57,17 +58,19 @@ function forEach<T>(sel: Selectable<T>, fn: (c: Channel<T>) => void): void {
 /* public methods */
 
 function channel<T>(): Channel<T> {
-  return {
+  const chan: Channel<T> = {
     [messages]: [],
     [putters]: [],
     [takers]: [],
     [racers]: [],
+    get length() { return chan[messages].length; },
```

Comment:
```suggestion
    get length() { return this[messages].length; },
```

if we use `this` here we can get rid of the of the redundant assignment of `chan` and keep the immediate return.

Reply:
Yo, that totally tracks. Also noticed I botched the text in my spec, so I fixed that as well.